[Doc] Graph Construction PR Refactor - PR2: GPartition Doc (#918)
awslabs/graphstorm: docs/source/graph-construction/gs-processing/gspartition/index.rst

.. _gspartition_index:

===================================
Running partition jobs on AWS Infra
===================================

> GraphStorm Distributed Data Partition (GSPartition) allows users to do distributed partition on preprocessed graph data

Author: Nope. But we use the term ``GraphStorm Distributed Data Processing`` for GSProcessing, shall we use the similar pattern ``GraphStorm Distributed Data Partition`` here?

> prepared by :ref:`GSProcessing<gsprocessing_prerequisites_index>`. To enable distributed training, the preprocessed input data must be converted to a partitioned graph representation.

Author: Let us use the gsprocessing-get-started page anchor here.

GSPartition allows user to handle massive graph data in distributed clusters on AWS. GSPartition is built on top of the
dgl `distributed graph partitioning pipeline <https://docs.dgl.ai/en/latest/guide/distributed-preprocessing.html#distributed-graph-partitioning-pipeline>`_.

> GSPartition consists of two steps: Graph Partitioning and Data Dispatching. Graph Partitioning step will calculate the ownership of

Reviewer: ```suggestion
GSPartition consists of two steps: Graph Partitioning and Data Dispatching. Graph Partitioning step will compute the ownership of
```

Author: Let us use ``calculate`` which as mentioned in the original DGL doc: https://docs.dgl.ai/en/latest/guide/distributed-preprocessing.html#distributed-graph-partitioning-

Reviewer 2: We can just say that the step assigns each node to one partition?

Author: I think it can make sense.

each partition and save the results as a set of files called partition assignment. Data Dispatching step will physically partitions the

> graph data and dispatch them to the machines user specified. It will generate the graph data in DGL format for distributed training.

Reviewer: ```suggestion
graph data and dispatch them according to the partition assignment. It will generate the graph data in DGL format for distributed training and inference.
```

> each partition and save the results as a set of files called partition assignment. Data Dispatching step will physically partitions the
> graph data and dispatch them according to the partition assignment. It will generate the graph data in DGL format for distributed training and inference.

Reviewer 2: ```suggestion
each partition and save the results as a set of files called partition assignment. Data Dispatching step will physically partition the
graph data and dispatch them according to the partition assignment. It will generate the graph data in DGL format, ready for distributed training and inference.
```
